### PR TITLE
Hack together some socket-activated goodness

### PR DESCRIPTION
**Pull request by bcwaldon**

### CODE REVIEW

File: app/handle.go [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+)
+
+func NewRouter(cfg *Config, mg *mailgun.Client) *mux.Router {
+	hdlr := NewHandler(cfg, mg)
+
+	r := mux.NewRouter()
+
+	r.HandleFunc("/{action}/{list}", hdlr.initialHandler)
+	r.HandleFunc("/{action}/{list}/confirm/{email}/{token}", hdlr.confirmationHandler)
+	r.HandleFunc("/health", hdlr.healthHandler)
```
Comment (philips) [inline app/handle.go:22]: OH YEA

File: app/mail.go [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+	bcc       []string
+	subject   string
+	html      string
+	text      string
+	headers   map[string]string
+	options   map[string]string
+	variables map[string]string
+}
+
+func (m *mail) From() string                 { return m.from }
```
Comment (philips) [inline app/mail.go:16]: Did I tell you how much I hate this mailgun library?

File: app/server.go [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+	ls, e := listenFD()
+	fmt.Printf("Got %d listeners", len(ls))
+	if e != nil {
+		println(e.Error())
+	}
+
+	chErrors := make(chan error, len(ls))
+
+	// Since listenFD will return one or more sockets we have
+	// to create a go func to spawn off multiple serves
```
Comment (philips) [inline app/server.go:27]: Isn't that the best? So silly.

File: app/handle.go [hunk 4]
```diff
@@ -0,0 +1,239 @@
...
+
+	if err != nil {
+		http.Error(w, "Internal error", 500)
+		fmt.Println(err)
+		return
+	}
+}
+
+func (h *Handler) healthHandler(w http.ResponseWriter, r *http.Request) {
+	fmt.Fprint(w, "OK")
```
Comment (philips) [inline app/handle.go:238]: :( dang. I guess I will have to do that. :-P

Comment (bcwaldon) [inline app/handle.go:238]: I just didn't get that far before walking out the door last night. Don't lose hope, yet!

Comment (philips) [inline app/handle.go:238]: haha

**Pull request merged**